improver/cube_combiner.py

```diff
@@ -277,6 +278,12 @@ def process(
         if self.operation == "mean":
             result.data = result.data / len(cube_list)
 
+        # Check resulting dtype and modify if necessary
+        types_to_fix = FLOAT_TYPES.copy()
+        types_to_fix.remove(FLOAT_DTYPE)
+        if result.dtype in types_to_fix:
```

Comment:
Could do e.g. `if result.dtype in FLOAT_TYPES and result.dtype != FLOAT_DTYPE`.

`FLOAT_DTYPE` maybe should be called something like `FLOAT_TYPE_OUTPUT` or `FLOAT_TYPE_DESIRED` if that's really what it is.